numaprom/watcher.py

```diff
@@ -0,0 +1,77 @@
+import os
+import time
+
+from omegaconf import OmegaConf
+from watchdog.observers import Observer
+from numalogic.config import NumalogicConf
+from watchdog.events import FileSystemEventHandler
+
+from numaprom._constants import CONFIG_DIR, DEFAULT_CONFIG_DIR
+from numaprom import NumapromConf, get_logger
+
+_LOGGER = get_logger(__name__)
+
+config = {}
+
+
+def load_configs():
+    schema: NumapromConf = OmegaConf.structured(NumapromConf)
+
+    conf = OmegaConf.load(os.path.join(CONFIG_DIR, "config.yaml"))
+    app_configs = OmegaConf.merge(schema, conf).configs
+
+    conf = OmegaConf.load(os.path.join(DEFAULT_CONFIG_DIR, "config.yaml"))
+    default_configs = OmegaConf.merge(schema, conf).configs
+
+    conf = OmegaConf.load(os.path.join(DEFAULT_CONFIG_DIR, "numalogic_config.yaml"))
+    schema: NumalogicConf = OmegaConf.structured(NumalogicConf)
+    default_numalogic = OmegaConf.merge(schema, conf)
+
+    return app_configs, default_configs, default_numalogic
+
+
+def update_configs():
```

Comment:
We should add the `global` keyword if we are referencing the `config` variable

Reply:
We don't have to use the global keyword if its defined in the same scope

Reply:
But what if someone accidentally adds another variable of the same name in the function? :) An alternative is to rename it something like `CONFIG`, so it is clear.